obspy/io/gcf/libgcf.py

```diff
@@ -0,0 +1,182 @@
+# reads Guralp Compressed Format (GCF) Files
+# By Ran Novitsky Nof @ BSL, 2016
+# ran.nof@gmail.com
+# Based on Guralp's GCF reference (GCF-RFC-GCFR, Issue C, 2011-01-05)
+# more details available from: http://www.guralp.com/apps/ok?doc=GCF_Intro
+# last access: June, 2016
+import numpy as np
+from obspy import UTCDateTime
+
+SPSD = {  # Table 3.1: special sample rates
+    157: 0.1,
+    161: 0.125,
+    162: 0.2,
+    164: 0.25,
+    167: 0.5,
+    171: 400,
+    174: 500,
+    176: 1000,
+    179: 2000,
+    181: 4000
+    }
+ToffsetsD = {  # Table 3.1: Time fractional offset denominator
+    171: 8.,
+    174: 2.,
+    176: 4.,
+    179: 8.,
+    181: 16.
+    }
+compressionD = {  # Table 3.2: format field to data type
+    1: '>i4',
+    2: '>i2',
+    4: '>i1'
+    }
+bitsD = {  # Table 3.2: format field to data bytes
+    1: 4,  # 32bits
+    2: 2,  # 16bits
+    4: 1   # 8bits
+    }
+
+
+def is_gcf(f):
+    'Test if file is GCF by reading at least 1 data block'
+    while True:
+        header, data = read(f)
+        if len(data):
+            break
+
+
+def decode36(DATA):
+    'Decode base 36 data'
+    # http://geophysics.eas.gatech.edu/GTEQ/Scream4.4/Decoding_Base_36_numbers_C.htm
+    STR = ''
+    while DATA:
+        imed = DATA % 36
+        if imed > 9:
+            imed += 7
+        STR = chr(imed + 48) + STR
+        DATA = DATA / 36
+    return STR
+
+
+def decodeTime(DATA):
+    'Decode time field'
+    H = DATA / 3600
+    M = (DATA % 3600) / 60
+    S = DATA % 60
+    return "%02d:%02d:%02d" % (H, M, S)
+
+
+def isLeapYear(Year):
+    return (Year % 4 == 0) and ((Year % 100 != 0) or (Year % 400 == 0))
+
+
+def decodeDate(days):
+    'Decode date field'
+    daycount = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
+    year = 1989
+    month = 10  # note that month and day are zero referenced
+    # The number of days is referenced from 17 Nov 1989
+    days += 16
+    # Account for leap years
+    if isLeapYear(year):
+        daycount[1] = 29
+    while days >= daycount[month]:
+        days = days - daycount[month]
+        year = year + (month + 1) / 12
+        month = (month + 1) % 12
+        if isLeapYear(year):
+            daycount[1] = 29
+        else:
+            daycount[1] = 28
+    return "%4d-%02d-%02d" % (year, month + 1, days + 1)
+
+
+def decodeDateTime(data):
+    '''Decode date and time field.
+    The date code is a 32 bit value specifying the start time of the block.
+    Bits 0-16 contain the number of seconds since midnight,
+    and bits 17-31 the number of days since 17th November 1989.'''
+    DATE = decodeDate(data >> 17)
+    TIME = decodeTime(data - (data >> 17 << 17))
+    starttime = UTCDateTime(DATE + 'T' + TIME)
+    return starttime
+
+
+def readDataBlock(f, skipData=False):
```

Comment:
:+1: Please definitely rename this option to `headonly` across the complete PR to be in line with other waveform plugins in obspy.